### PR TITLE
Add transformative redirects to finder frontend

### PR DESCRIPTION
**Pull request by callumknights**
This PR adds both the consultations and statistics transformative redirects that were previously handled by whitehall into finder-frontend.

These need to be handled differently to other redirects being moved from Whitehall as there is some setting of query string parameters currently done by whitehall, as well as transformation from legacy params to newer ones. 

This is done as part of the mission to remove any frontend rendering form whitehall, as these redirects are handled by whitehall frontend instances, they needed to be moved. 

Trello: https://trello.com/c/SmdAt9SX/523-migrate-redirect-logic-from-whitehall-to-finder-frontend, [Jira issue PP-716](https://gov-uk.atlassian.net/browse/PP-716)

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

Follow [these steps](https://guides.rubyonrails.org/upgrading_ruby_on_rails.html) if you are doing a Rails upgrade.

### CODE REVIEW

File: app/controllers/redirection_controller.rb [hunk 1]
```diff
@@ -16,6 +16,22 @@ def redirect_latest
...
+  def redirect_consultations
+    topics = {}
+    topics["level_one_taxon"] = params[:topics] if params[:topics]
+    topics["level_two_taxon"] = params[:subtaxons] if params[:subtaxons]
+
+    redirect_params = params.merge(topics)
+                            .slice(:departments, :level_one_taxon, :level_two_taxon, :world_locations, :content_store_document_type)
+                            .permit(:level_one_taxon, :level_two_taxon, departments: [],  world_locations: [], content_store_document_type: [])
+                            .transform_keys { |k| k == "departments" ? "organisations" : k }
+                            # .transform_values { |v| v == (["open_consultations"] || ["closed_consultations"] || ["policy_papers"]) ? %w[open_consultations closed_consultations] : v }
```
Comment (brucebolt) [inline app/controllers/redirection_controller.rb]: You probably want to delete this commented line from the first commit.

File: app/controllers/redirection_controller.rb [hunk 2]
```diff
@@ -16,6 +16,22 @@ def redirect_latest
...
   end
 
+  def redirect_consultations
+    topics = {}
+    topics["level_one_taxon"] = params[:topics] if params[:topics]
+    topics["level_two_taxon"] = params[:subtaxons] if params[:subtaxons]
+
+    redirect_params = params.merge(topics)
+                            .slice(:departments, :level_one_taxon, :level_two_taxon, :world_locations, :content_store_document_type)
+                            .permit(:level_one_taxon, :level_two_taxon, departments: [],  world_locations: [], content_store_document_type: [])
```
Comment (brucebolt) [inline app/controllers/redirection_controller.rb]: There's two spaces between the end of departments array and world locations.  It's fixed in the next commit, but good to do that here too.

**Review by brucebolt — APPROVED** [review]
Looks good to me, especially since we've discussed a lot of this on Slack already 👍.

Just added a couple of comments about tidying up the first commit (since there's a bit of extra diff in both commits that's probably not needed).

File: app/controllers/redirection_controller.rb (hunk 1 above)
Comment (callumknights) [reply app/controllers/redirection_controller.rb]: 👍

File: app/controllers/redirection_controller.rb (hunk 2 above)
Comment (callumknights) [reply app/controllers/redirection_controller.rb]: 👍

**Pull request merged**